src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -0,0 +1,172 @@
+    useBackgroundBlur();
+  const [isVideoTransformCheckBoxOn, setisVideoTransformCheckBoxOn] =
+    useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [
+    dropdownWithVideoTransformOptions,
+    setDropdownWithVideoTransformOptions,
+  ] = useState<ReactNode[] | null>(null);
+  const [activeVideoDevice, setDevice] = useState<
+    Device | VideoTransformDevice | null
+  >(meetingManager.selectedVideoInputDevice);
+
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
+  useEffect(() => {
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    return () => {
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    const deviceOptions: ReactNode[] = devices.map((option) => (
+      <PopOverItem
+        key={option.deviceId}
+        children={<span>{option.label}</span>}
+        checked={isOptionActive(selectedDevice, option.deviceId)}
+        onClick={async (): Promise<void> => {
+          // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
+          // If background blur is not on, then do a normal video selection
+          if (isVideoTransformDevice(activeVideoDevice) && !isLoading) {
+            setIsLoading(true);
+            const receivedDevice = videoInputSelectionToDevice(option.deviceId);
+            if (activeVideoDevice instanceof DefaultVideoTransformDevice) {
+              const transformedDevice =
+                activeVideoDevice.chooseNewInnerDevice(receivedDevice);
+              await meetingManager.selectVideoInputDevice(transformedDevice);
+            } else {
+              console.error('Transform device cannot choose new inner device');
+            }
+            setIsLoading(false);
+          } else {
+            await meetingManager.selectVideoInputDevice(option.deviceId);
+          }
+        }}
+      />
+    ));
+    if (isBackgroundBlurSupported) {
+      const videoTransformOptions: ReactNode = (
+        <PopOverItem
+          key="videoinput"
+          children={
+            <>
+              {isLoading && <Spinner width="1.5rem" height="1.5rem" />}
+              {isVideoTransformDevice(activeVideoDevice)
+                ? backgroundBlurOnLabel
+                : backgroundBlurOffLabel}
+            </>
+          }
+          checked={isVideoTransformDevice(activeVideoDevice)}
+          disabled={isLoading}
+          onClick={() => {
+            setIsLoading(true);
+            setisVideoTransformCheckBoxOn((current) => !current);
+          }}
+        />
+      );
+      deviceOptions.push(<PopOverSeparator key="separator" />);
+      deviceOptions.push(videoTransformOptions);
+    }
+    setDropdownWithVideoTransformOptions(deviceOptions);
+  }, [
+    createBackgroundBlurDevice,
+    activeVideoDevice,
+    devices.length,
```

Comment:
This devices.length has been a problem for the audio input, should we do something similar her as done in the below PR:
https://github.com/aws/amazon-chime-sdk-component-library-react/pull/644

Reply:
Yeah but in term of coding, it is better to do so than just compare based on devices.length right? Also consistency :D

Reply:
okay, will do the same as audioinputvfcontrol then